scipy: 1.9.3 -> 1.10.1

###### Checklist

<!-- Note that these are hard requirements -->

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name

`nix search` name: scipy
current version: 1.9.3
desired version: 1.10.1

###### Notify maintainers

maintainers: @FRidh

###### Note for maintainers

Please tag this issue in your PR.

##### My attempt

Some test data should be downloaded in advance. This is what I tried (on branch `master`):

```patch
commit 10f9e2bde6681023f6c7427a6e2e6aa082cc73d9
Author: Doron Behar <doron.behar@gmail.com>
Date:   Sat Jan 7 09:03:50 2023 +0200

    python3.pkgs.scipy: 1.9.3 -> 1.10.0

diff --git a/pkgs/development/python-modules/scipy/default.nix b/pkgs/development/python-modules/scipy/default.nix
index bec9752bc44..3e02c3866b4 100644
--- a/pkgs/development/python-modules/scipy/default.nix
+++ b/pkgs/development/python-modules/scipy/default.nix
@@ -13,6 +13,7 @@
 , nose
 , pytest
 , pytest-xdist
+, pooch
 , numpy
 , pybind11
 , libxcrypt
@@ -20,12 +21,12 @@
 
 buildPythonPackage rec {
   pname = "scipy";
-  version = "1.9.3";
+  version = "1.10.0";
   format = "pyproject";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-+8XAXIXBoCvnex/1kQh8g7xEV5xtK9n7eYu2TqXhoCc=";
+    sha256 = "sha256-yLPLxjaoeom3cMavyZm6pry7AWkbXMu8GxeRx8CgdUA=";
   };
 
   nativeBuildInputs = [ cython gfortran meson-python pythran pkg-config wheel ];
@@ -39,7 +40,7 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [ numpy ];
 
-  checkInputs = [ nose pytest pytest-xdist ];
+  checkInputs = [ nose pytest pytest-xdist pooch ];
 
   doCheck = !(stdenv.isx86_64 && stdenv.isDarwin);
 
```

And tests failed with:

<details>

```
==================================== ERRORS ====================================
__________________ ERROR at setup of TestDatasets.test_ascent __________________
[gw6] linux -- Python 3.10.9 /nix/store/al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/bin/python3.10
../gkisb2mbs8s0k2yvi0064y03r50c92hc-python3.10-pooch-1.6.0/lib/python3.10/site-packages/pooch/utils.py:249: in make_local_storage
    os.makedirs(path, exist_ok=True)
        action     = 'create'
        env        = None
        message    = ["[Errno 13] Permission denied: '/homeless-shelter'",
 '| Pooch could not create data cache folder '
 "'/homeless-shelter/.cache/scipy-data'.",
 'Will not be able to download data files.']
        path       = '/homeless-shelter/.cache/scipy-data'
../al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/os.py:215: in makedirs
    makedirs(head, exist_ok=exist_ok)
        exist_ok   = True
        head       = '/homeless-shelter/.cache'
        mode       = 511
        name       = '/homeless-shelter/.cache/scipy-data'
        tail       = 'scipy-data'
../al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/os.py:215: in makedirs
    makedirs(head, exist_ok=exist_ok)
        exist_ok   = True
        head       = '/homeless-shelter'
        mode       = 511
        name       = '/homeless-shelter/.cache'
        tail       = '.cache'
../al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/os.py:225: in makedirs
    mkdir(name, mode)
E   PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
        exist_ok   = True
        head       = '/'
        mode       = 511
        name       = '/homeless-shelter'
        tail       = 'homeless-shelter'

The above exception was the direct cause of the following exception:
lib/python3.10/site-packages/scipy/datasets/tests/test_data.py:34: in test_download_all
    download_all()
        self       = <scipy.datasets.tests.test_data.TestDatasets object at 0x7fffc68d9420>
lib/python3.10/site-packages/scipy/datasets/_download_all.py:42: in download_all
    pooch.retrieve(url=_registry.registry_urls[dataset_name],
        dataset_hash = '03ce124c1afc880f87b55f6b061110e2e1e939679184f5614e38dacc6c1957e2'
        dataset_name = 'ascent.dat'
        path       = PosixPath('/homeless-shelter/.cache/scipy-data')
../gkisb2mbs8s0k2yvi0064y03r50c92hc-python3.10-pooch-1.6.0/lib/python3.10/site-packages/pooch/core.py:224: in retrieve
    make_local_storage(path)
        downloader = None
        fname      = 'ascent.dat'
        known_hash = '03ce124c1afc880f87b55f6b061110e2e1e939679184f5614e38dacc6c1957e2'
        path       = PosixPath('/homeless-shelter/.cache/scipy-data')
        processor  = None
        progressbar = False
        url        = 'https://raw.githubusercontent.com/scipy/dataset-ascent/main/ascent.dat'
../gkisb2mbs8s0k2yvi0064y03r50c92hc-python3.10-pooch-1.6.0/lib/python3.10/site-packages/pooch/utils.py:264: in make_local_storage
    raise PermissionError(" ".join(message)) from error
E   PermissionError: [Errno 13] Permission denied: '/homeless-shelter' | Pooch could not create data cache folder '/homeless-shelter/.cache/scipy-data'. Will not be able to download data files.
        action     = 'create'
        env        = None
        message    = ["[Errno 13] Permission denied: '/homeless-shelter'",
 '| Pooch could not create data cache folder '
 "'/homeless-shelter/.cache/scipy-data'.",
 'Will not be able to download data files.']
        path       = '/homeless-shelter/.cache/scipy-data'
______________ ERROR at setup of TestDatasets.test_existence_all _______________
[gw5] linux -- Python 3.10.9 /nix/store/al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/bin/python3.10
../gkisb2mbs8s0k2yvi0064y03r50c92hc-python3.10-pooch-1.6.0/lib/python3.10/site-packages/pooch/utils.py:249: in make_local_storage
    os.makedirs(path, exist_ok=True)
        action     = 'create'
        env        = None
        message    = ["[Errno 13] Permission denied: '/homeless-shelter'",
 '| Pooch could not create data cache folder '
 "'/homeless-shelter/.cache/scipy-data'.",
 'Will not be able to download data files.']
        path       = '/homeless-shelter/.cache/scipy-data'
../al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/os.py:215: in makedirs
    makedirs(head, exist_ok=exist_ok)
        exist_ok   = True
        head       = '/homeless-shelter/.cache'
        mode       = 511
        name       = '/homeless-shelter/.cache/scipy-data'
        tail       = 'scipy-data'
../al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/os.py:215: in makedirs
    makedirs(head, exist_ok=exist_ok)
        exist_ok   = True
        head       = '/homeless-shelter'
        mode       = 511
        name       = '/homeless-shelter/.cache'
        tail       = '.cache'
../al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/os.py:225: in makedirs
    mkdir(name, mode)
E   PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
        exist_ok   = True
        head       = '/'
        mode       = 511
        name       = '/homeless-shelter'
        tail       = 'homeless-shelter'

The above exception was the direct cause of the following exception:
lib/python3.10/site-packages/scipy/datasets/tests/test_data.py:34: in test_download_all
    download_all()
        self       = <scipy.datasets.tests.test_data.TestDatasets object at 0x7fffc68d9420>
lib/python3.10/site-packages/scipy/datasets/_download_all.py:42: in download_all
    pooch.retrieve(url=_registry.registry_urls[dataset_name],
        dataset_hash = '03ce124c1afc880f87b55f6b061110e2e1e939679184f5614e38dacc6c1957e2'
        dataset_name = 'ascent.dat'
        path       = PosixPath('/homeless-shelter/.cache/scipy-data')
../gkisb2mbs8s0k2yvi0064y03r50c92hc-python3.10-pooch-1.6.0/lib/python3.10/site-packages/pooch/core.py:224: in retrieve
    make_local_storage(path)
        downloader = None
        fname      = 'ascent.dat'
        known_hash = '03ce124c1afc880f87b55f6b061110e2e1e939679184f5614e38dacc6c1957e2'
        path       = PosixPath('/homeless-shelter/.cache/scipy-data')
        processor  = None
        progressbar = False
        url        = 'https://raw.githubusercontent.com/scipy/dataset-ascent/main/ascent.dat'
../gkisb2mbs8s0k2yvi0064y03r50c92hc-python3.10-pooch-1.6.0/lib/python3.10/site-packages/pooch/utils.py:264: in make_local_storage
    raise PermissionError(" ".join(message)) from error
E   PermissionError: [Errno 13] Permission denied: '/homeless-shelter' | Pooch could not create data cache folder '/homeless-shelter/.cache/scipy-data'. Will not be able to download data files.
        action     = 'create'
        env        = None
        message    = ["[Errno 13] Permission denied: '/homeless-shelter'",
 '| Pooch could not create data cache folder '
 "'/homeless-shelter/.cache/scipy-data'.",
 'Will not be able to download data files.']
        path       = '/homeless-shelter/.cache/scipy-data'
____________ ERROR at setup of TestDatasets.test_electrocardiogram _____________
[gw0] linux -- Python 3.10.9 /nix/store/al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/bin/python3.10
../gkisb2mbs8s0k2yvi0064y03r50c92hc-python3.10-pooch-1.6.0/lib/python3.10/site-packages/pooch/utils.py:249: in make_local_storage
    os.makedirs(path, exist_ok=True)
        action     = 'create'
        env        = None
        message    = ["[Errno 13] Permission denied: '/homeless-shelter'",
 '| Pooch could not create data cache folder '
 "'/homeless-shelter/.cache/scipy-data'.",
 'Will not be able to download data files.']
        path       = '/homeless-shelter/.cache/scipy-data'
../al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/os.py:215: in makedirs
    makedirs(head, exist_ok=exist_ok)
        exist_ok   = True
        head       = '/homeless-shelter/.cache'
        mode       = 511
        name       = '/homeless-shelter/.cache/scipy-data'
        tail       = 'scipy-data'
../al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/os.py:215: in makedirs
    makedirs(head, exist_ok=exist_ok)
        exist_ok   = True
        head       = '/homeless-shelter'
        mode       = 511
        name       = '/homeless-shelter/.cache'
        tail       = '.cache'
../al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/os.py:225: in makedirs
    mkdir(name, mode)
E   PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
        exist_ok   = True
        head       = '/'
        mode       = 511
        name       = '/homeless-shelter'
        tail       = 'homeless-shelter'

The above exception was the direct cause of the following exception:
lib/python3.10/site-packages/scipy/datasets/tests/test_data.py:34: in test_download_all
    download_all()
        self       = <scipy.datasets.tests.test_data.TestDatasets object at 0x7fffc6baa650>
lib/python3.10/site-packages/scipy/datasets/_download_all.py:42: in download_all
    pooch.retrieve(url=_registry.registry_urls[dataset_name],
        dataset_hash = '03ce124c1afc880f87b55f6b061110e2e1e939679184f5614e38dacc6c1957e2'
        dataset_name = 'ascent.dat'
        path       = PosixPath('/homeless-shelter/.cache/scipy-data')
../gkisb2mbs8s0k2yvi0064y03r50c92hc-python3.10-pooch-1.6.0/lib/python3.10/site-packages/pooch/core.py:224: in retrieve
    make_local_storage(path)
        downloader = None
        fname      = 'ascent.dat'
        known_hash = '03ce124c1afc880f87b55f6b061110e2e1e939679184f5614e38dacc6c1957e2'
        path       = PosixPath('/homeless-shelter/.cache/scipy-data')
        processor  = None
        progressbar = False
        url        = 'https://raw.githubusercontent.com/scipy/dataset-ascent/main/ascent.dat'
../gkisb2mbs8s0k2yvi0064y03r50c92hc-python3.10-pooch-1.6.0/lib/python3.10/site-packages/pooch/utils.py:264: in make_local_storage
    raise PermissionError(" ".join(message)) from error
E   PermissionError: [Errno 13] Permission denied: '/homeless-shelter' | Pooch could not create data cache folder '/homeless-shelter/.cache/scipy-data'. Will not be able to download data files.
        action     = 'create'
        env        = None
        message    = ["[Errno 13] Permission denied: '/homeless-shelter'",
 '| Pooch could not create data cache folder '
 "'/homeless-shelter/.cache/scipy-data'.",
 'Will not be able to download data files.']
        path       = '/homeless-shelter/.cache/scipy-data'
___________________ ERROR at setup of TestDatasets.test_face ___________________
[gw7] linux -- Python 3.10.9 /nix/store/al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/bin/python3.10
../gkisb2mbs8s0k2yvi0064y03r50c92hc-python3.10-pooch-1.6.0/lib/python3.10/site-packages/pooch/utils.py:249: in make_local_storage
    os.makedirs(path, exist_ok=True)
        action     = 'create'
        env        = None
        message    = ["[Errno 13] Permission denied: '/homeless-shelter'",
 '| Pooch could not create data cache folder '
 "'/homeless-shelter/.cache/scipy-data'.",
 'Will not be able to download data files.']
        path       = '/homeless-shelter/.cache/scipy-data'
../al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/os.py:215: in makedirs
    makedirs(head, exist_ok=exist_ok)
        exist_ok   = True
        head       = '/homeless-shelter/.cache'
        mode       = 511
        name       = '/homeless-shelter/.cache/scipy-data'
        tail       = 'scipy-data'
../al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/os.py:215: in makedirs
    makedirs(head, exist_ok=exist_ok)
        exist_ok   = True
        head       = '/homeless-shelter'
        mode       = 511
        name       = '/homeless-shelter/.cache'
        tail       = '.cache'
../al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/os.py:225: in makedirs
    mkdir(name, mode)
E   PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
        exist_ok   = True
        head       = '/'
        mode       = 511
        name       = '/homeless-shelter'
        tail       = 'homeless-shelter'

The above exception was the direct cause of the following exception:
lib/python3.10/site-packages/scipy/datasets/tests/test_data.py:34: in test_download_all
    download_all()
        self       = <scipy.datasets.tests.test_data.TestDatasets object at 0x7fffc68d9420>
lib/python3.10/site-packages/scipy/datasets/_download_all.py:42: in download_all
    pooch.retrieve(url=_registry.registry_urls[dataset_name],
        dataset_hash = '03ce124c1afc880f87b55f6b061110e2e1e939679184f5614e38dacc6c1957e2'
        dataset_name = 'ascent.dat'
        path       = PosixPath('/homeless-shelter/.cache/scipy-data')
../gkisb2mbs8s0k2yvi0064y03r50c92hc-python3.10-pooch-1.6.0/lib/python3.10/site-packages/pooch/core.py:224: in retrieve
    make_local_storage(path)
        downloader = None
        fname      = 'ascent.dat'
        known_hash = '03ce124c1afc880f87b55f6b061110e2e1e939679184f5614e38dacc6c1957e2'
        path       = PosixPath('/homeless-shelter/.cache/scipy-data')
        processor  = None
        progressbar = False
        url        = 'https://raw.githubusercontent.com/scipy/dataset-ascent/main/ascent.dat'
../gkisb2mbs8s0k2yvi0064y03r50c92hc-python3.10-pooch-1.6.0/lib/python3.10/site-packages/pooch/utils.py:264: in make_local_storage
    raise PermissionError(" ".join(message)) from error
E   PermissionError: [Errno 13] Permission denied: '/homeless-shelter' | Pooch could not create data cache folder '/homeless-shelter/.cache/scipy-data'. Will not be able to download data files.
        action     = 'create'
        env        = None
        message    = ["[Errno 13] Permission denied: '/homeless-shelter'",
 '| Pooch could not create data cache folder '
 "'/homeless-shelter/.cache/scipy-data'.",
 'Will not be able to download data files.']
        path       = '/homeless-shelter/.cache/scipy-data'
=============================== warnings summary ===============================
lib/python3.10/site-packages/scipy/optimize/tests/test_linprog.py: 10 warnings
lib/python3.10/site-packages/scipy/optimize/tests/test__linprog_clean_inputs.py: 2 warnings
  /nix/store/cyndlxq2pdjwxm94yq52hd6v1lnv7jgs-python3.10-numpy-1.23.3/lib/python3.10/site-packages/numpy/core/numeric.py:2457: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
    a1, a2 = asarray(a1), asarray(a2)

lib/python3.10/site-packages/scipy/optimize/tests/test_minimize_constrained.py::TestBoundedNelderMead::test_outside_bounds_warning
  /nix/store/b3n33phksn2k021wbck835s3gvshnfrj-python3.10-scipy-1.10.0/lib/python3.10/site-packages/scipy/optimize/tests/test_minimize_constrained.py:779: OptimizeWarning: Initial guess is not within the specified bounds
    minimize(prob.fun, [-10, 8],

lib/python3.10/site-packages/scipy/sparse/linalg/_isolve/tests/test_iterative.py::test_maxiter_worsening[tfqmr]
  /nix/store/b3n33phksn2k021wbck835s3gvshnfrj-python3.10-scipy-1.10.0/lib/python3.10/site-packages/scipy/sparse/linalg/_isolve/tfqmr.py:148: RuntimeWarning: divide by zero encountered in cdouble_scalars
    d = u + (theta**2 / alpha) * eta * d  # [1]-(5.5)

lib/python3.10/site-packages/scipy/sparse/linalg/_isolve/tests/test_iterative.py::test_maxiter_worsening[tfqmr]
  /nix/store/b3n33phksn2k021wbck835s3gvshnfrj-python3.10-scipy-1.10.0/lib/python3.10/site-packages/scipy/sparse/linalg/_isolve/tfqmr.py:148: RuntimeWarning: invalid value encountered in cdouble_scalars
    d = u + (theta**2 / alpha) * eta * d  # [1]-(5.5)

lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py: 15 warnings
  /nix/store/cyndlxq2pdjwxm94yq52hd6v1lnv7jgs-python3.10-numpy-1.23.3/lib/python3.10/site-packages/numpy/lib/function_base.py:518: RuntimeWarning: Mean of empty slice.
    avg = a.mean(axis, **keepdims_kw)

lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[empty-propagate-kurtosis-args9-kwds9-1-1-False-<lambda>]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[empty-omit-kurtosis-args9-kwds9-1-1-False-<lambda>]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[empty-raise-kurtosis-args9-kwds9-1-1-False-<lambda>]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-kurtosis-args9-kwds9-1-1-False-<lambda>]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[all_nans-omit-kurtosis-args9-kwds9-1-1-False-<lambda>]
  /nix/store/b3n33phksn2k021wbck835s3gvshnfrj-python3.10-scipy-1.10.0/lib/python3.10/site-packages/scipy/stats/_stats_py.py:1417: RuntimeWarning: Mean of empty slice.
    mean = a.mean(axis, keepdims=True)

lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py: 25 warnings
  /nix/store/cyndlxq2pdjwxm94yq52hd6v1lnv7jgs-python3.10-numpy-1.23.3/lib/python3.10/site-packages/numpy/core/fromnumeric.py:3432: RuntimeWarning: Mean of empty slice.
    return _methods._mean(a, axis=axis, dtype=dtype,

lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[empty-propagate-skew-args10-kwds10-1-1-False-<lambda>]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[empty-omit-skew-args10-kwds10-1-1-False-<lambda>]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[empty-raise-skew-args10-kwds10-1-1-False-<lambda>]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-skew-args10-kwds10-1-1-False-<lambda>]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[all_nans-omit-skew-args10-kwds10-1-1-False-<lambda>]
  /nix/store/b3n33phksn2k021wbck835s3gvshnfrj-python3.10-scipy-1.10.0/lib/python3.10/site-packages/scipy/stats/_stats_py.py:1304: RuntimeWarning: Mean of empty slice.
    mean = a.mean(axis, keepdims=True)

lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[empty-propagate-moment-args14-kwds14-1-2-False-None]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[empty-omit-moment-args14-kwds14-1-2-False-None]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[empty-raise-moment-args14-kwds14-1-2-False-None]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-moment-args14-kwds14-1-2-False-None]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[all_nans-omit-moment-args14-kwds14-1-2-False-None]
  /nix/store/b3n33phksn2k021wbck835s3gvshnfrj-python3.10-scipy-1.10.0/lib/python3.10/site-packages/scipy/stats/_stats_py.py:1141: RuntimeWarning: Mean of empty slice.
    mean = a.mean(axis, keepdims=True)

lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-wilcoxon-args5-kwds5-1-3-True-<lambda>]
  /nix/store/b3n33phksn2k021wbck835s3gvshnfrj-python3.10-scipy-1.10.0/lib/python3.10/site-packages/scipy/stats/_morestats.py:3428: UserWarning: Sample size too small for normal approximation.
    warnings.warn("Sample size too small for normal approximation.")

lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-kurtosis-args9-kwds9-1-1-False-<lambda>]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-skew-args10-kwds10-1-1-False-<lambda>]
  /nix/store/b3n33phksn2k021wbck835s3gvshnfrj-python3.10-scipy-1.10.0/lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py:160: RuntimeWarning: Precision loss occurred in moment calculation due to catastrophic cancellation. This occurs when the data are nearly identical. Results may be unreliable.
    return unpacker(hypotest(*data1d, *args, _no_deco=_no_deco, **kwds))

lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-kurtosis-args9-kwds9-1-1-False-<lambda>]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-skew-args10-kwds10-1-1-False-<lambda>]
  /nix/store/b3n33phksn2k021wbck835s3gvshnfrj-python3.10-scipy-1.10.0/lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py:244: RuntimeWarning: Precision loss occurred in moment calculation due to catastrophic cancellation. This occurs when the data are nearly identical. Results may be unreliable.
    res1db = unpacker(hypotest(*data1d, *args,

lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-moment-args14-kwds14-1-2-False-None]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-ttest_1samp-args16-kwds16-1-7-False-unpack_ttest_result]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-ttest_rel-args17-kwds17-2-7-True-unpack_ttest_result]
  /nix/store/b3n33phksn2k021wbck835s3gvshnfrj-python3.10-scipy-1.10.0/lib/python3.10/site-packages/scipy/stats/_axis_nan_policy.py:376: RuntimeWarning: Precision loss occurred in moment calculation due to catastrophic cancellation. This occurs when the data are nearly identical. Results may be unreliable.
    return hypotest_fun_in(*args, **kwds)

lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-moment-args14-kwds14-1-2-False-None]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-ttest_1samp-args16-kwds16-1-7-False-unpack_ttest_result]
lib/python3.10/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-ttest_rel-args17-kwds17-2-7-True-unpack_ttest_result]
  /nix/store/b3n33phksn2k021wbck835s3gvshnfrj-python3.10-scipy-1.10.0/lib/python3.10/site-packages/scipy/stats/_axis_nan_policy.py:502: RuntimeWarning: Precision loss occurred in moment calculation due to catastrophic cancellation. This occurs when the data are nearly identical. Results may be unreliable.
    res = hypotest_fun_out(*samples, **kwds)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
```

</details>

--- a/pkgs/development/python-modules/scipy/default.nix
+++ b/pkgs/development/python-modules/scipy/default.nix
@@ -15,24 +15,33 @@
 , pytest-xdist
 , numpy
 , pybind11
+, pooch
 , libxcrypt
 }:
 
 buildPythonPackage rec {
   pname = "scipy";
-  version = "1.9.3";
+  version = "1.10.1";
   format = "pyproject";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-+8XAXIXBoCvnex/1kQh8g7xEV5xtK9n7eYu2TqXhoCc=";
+    sha256 = "sha256-LPnfuAp7RYm6TEDOdYiYbW1c68VFfK0sKID2vC1C86U=";
   };
+
+  patches = [
+    # These tests require internet connection, currently impossible to disable
+    # them otherwise, see:
+    # https://github.com/scipy/scipy/pull/17965
+    ./disable-datasets-tests.patch
+  ];
 
   nativeBuildInputs = [ cython gfortran meson-python pythran pkg-config wheel ];
 
   buildInputs = [
     numpy.blas
     pybind11
+    pooch
   ] ++ lib.optionals (pythonOlder "3.9") [
     libxcrypt
   ];

--- a/pkgs/development/python-modules/scipy/disable-datasets-tests.patch
+++ b/pkgs/development/python-modules/scipy/disable-datasets-tests.patch
@@ -1,0 +1,9 @@
+diff --git i/scipy/datasets/meson.build w/scipy/datasets/meson.build
+index 101377253..eec2feea4 100644
+--- i/scipy/datasets/meson.build
++++ w/scipy/datasets/meson.build
+@@ -11,4 +11,3 @@ py3.install_sources(
+   subdir: 'scipy/datasets'
+ )
+ 
+-subdir('tests')

